api/resource_groups_azure.go

```diff
@@ -0,0 +1,93 @@
+//
+// Author:: Darren Murray (<darren.murray@lacework.net>)
+// Copyright:: Copyright 2021, Lacework Inc.
+// License:: Apache License, Version 2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package api
+
+import "encoding/json"
+
+// GetAzureResourceGroup gets a single Azure ResourceGroup matching the
+// provided resource guid
+func (svc *ResourceGroupsService) GetAzureResourceGroup(guid string) (
+	response AzureResourceGroupResponse,
+	err error,
+) {
+	var rawResponse ResourceGroupResponse
+	err = svc.get(guid, &rawResponse)
+	if err != nil {
+		return AzureResourceGroupResponse{}, err
+	}
+
+	return setAzureResponse(rawResponse)
+}
+
+// UpdateAzureResourceGroup updates a single Azure ResourceGroup on the Lacework Server
+func (svc *ResourceGroupsService) UpdateAzureResourceGroup(data ResourceGroup) (
+	response AzureResourceGroupResponse,
+	err error,
+) {
+	var rawResponse ResourceGroupResponse
+	err = svc.update(data.ID(), data, &rawResponse)
+	if err != nil {
+		return AzureResourceGroupResponse{}, err
+	}
+
+	return setAzureResponse(rawResponse)
+}
+
+func setAzureResponse(response ResourceGroupResponse) (az AzureResourceGroupResponse, err error) {
+	var props AzureResourceGroupProps
+	az = AzureResourceGroupResponse{
+		Data: AzureResourceGroupData{
+			Guid:         response.Data.Guid,
+			IsDefault:    response.Data.IsDefault,
+			ResourceGuid: response.Data.ResourceGuid,
+			Name:         response.Data.Name,
+			Type:         response.Data.Type,
+			Enabled:      response.Data.Enabled,
+		},
+	}
+
+	err = json.Unmarshal([]byte(response.Data.Props.(string)), &props)
```

Comment:
.... Could we please replicate my suggestion in all other Resource Group Types? ... 😅